src/notebook/notebookFeature.ts

```diff
@@ -74,6 +122,38 @@ export class JuliaNotebookFeature {
             await currentKernel.queueCell(cell)
         }
     }
+    private getKernelSpecNameAndVersion(notebook: vscode.NotebookDocument): {name:string; version: string} {
+        const metadata = (notebook.metadata as JupyterNotebookMetadata)?.custom.metadata;
+        const kernelspecName = metadata?.kernelspec?.name || '';
+        const version = metadata?.language_info?.version || '';
+        return this.isJuliaNotebook(notebook) ? { name: kernelspecName, version } : {name:'', version:''}
+    }
+    private updateNotebookWithSelectedKernel(notebook: vscode.NotebookDocument, name: string, version: string) {
+        // Dont edit in place, create a copy of the metadata.
+        const nbmetadata: JupyterNotebookMetadata = JSON.parse(JSON.stringify((notebook.metadata || { custom: { metadata: {}}})));
+        nbmetadata.custom.metadata.kernelspec = {
+            display_name: name,
+            language: 'julia',
+            name: name
+        }
+        nbmetadata.custom.metadata.language_info = {
+            name: name,
+            version: version,
+            mimetype: 'text/julia',
+            file_extension: '.jl'
+        }
+        // TODO: Update the notebook metadata (when its stable).
```

Comment:
The API used to udpate the notebook metadata is still proposed API.
Without this, any notebook created will not have the julia kernel information in the metadata.
We can make use of the proposed API, however it will not work in stable.

Reply:
Let me check with the team

Reply:
Does "not work" mean it will crash, or it will just not save things properly? I think we should keep `master` in a state now where we can tag insider builds and ship them to our normal user base without breaking them.

Reply:
Oh, and this is something that has to ship in VS Code itself, right? So we are tied with this to the VS Code release cycle, not the Jupyter extension cycle?